### PR TITLE
Fix some req don't fill in collection id

### PR DESCRIPTION
**Pull request by smellthemoon**
Signed-off-by: lixinguo [xinguo.li@zilliz.com](mailto:xinguo.li@zilliz.com)
#23113

### CODE REVIEW

File: internal/querynodev2/services.go [hunk 1]
```diff
@@ -1006,9 +1006,10 @@ func (node *QueryNode) SyncDistribution(ctx context.Context, req *querypb.SyncDi
 
 	for nodeID, infos := range addSegments {
 		err := shardDelegator.LoadSegments(ctx, &querypb.LoadSegmentsRequest{
-			Base:      req.GetBase(),
-			DstNodeID: nodeID,
-			Infos:     infos,
+			Base:         req.GetBase(),
+			DstNodeID:    nodeID,
+			Infos:        infos,
+			CollectionID: req.CollectionID,
```
Comment (yah01) [inline internal/querynodev2/services.go:1012]: prefer `req.GetCollectionID()`

File: internal/querynodev2/services.go [hunk 2]
```diff
@@ -1017,9 +1018,10 @@ func (node *QueryNode) SyncDistribution(ctx context.Context, req *querypb.SyncDi
 
 	for _, action := range removeActions {
 		shardDelegator.ReleaseSegments(ctx, &querypb.ReleaseSegmentsRequest{
-			NodeID:     action.GetNodeID(),
-			SegmentIDs: []int64{action.GetSegmentID()},
-			Scope:      querypb.DataScope_Historical,
+			NodeID:       action.GetNodeID(),
+			SegmentIDs:   []int64{action.GetSegmentID()},
+			Scope:        querypb.DataScope_Historical,
+			CollectionID: action.GetInfo().GetCollectionID(),
```
Comment (yah01) [inline internal/querynodev2/services.go:1024]: For release action, the `Info` is `nil`

**Review by yah01 — CHANGES_REQUESTED** [review]

**Review by congqixia — APPROVED** [review]
/lgtm

**Comment by yah01** [comment]
is there the same problem with 2.2.0 branch? @smellthemoon

**Comment by smellthemoon** [comment]
> is there the same problem with 2.2.0 branch? @smellthemoon

No, this bug was introduced by qnv2.

**Pull request merged**